hbase-server/src/test/java/org/apache/hadoop/hbase/client/AbstractTestUpdateConfiguration.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.client;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.util.JVMClusterUtil.RegionServerThread;
+
+/**
+ * Base class to test Configuration Update logic.
+ */
+public abstract class AbstractTestUpdateConfiguration {
+  private static final int SERVER_CONFIG_INDEX  = 0;
```

Comment:
Why track indices when you could just have three variables like

```java
private static final String SERVER_CONFIG = "hbase-site.xml";
private static final String OVERWRITE_SERVER_CONFIG = "hbase-site2.xml";
...
```

This is what I meant in my earlier comment about descriptive variable names.

And for easier debugging, why call the file names 1, 2, 3 -- they can be named whatever we want since we're manually adding them as resources. Might as well make the file names descriptive too, so someone doing forensics later can have a clue. Something like "copy-of-hbase-site.xml", "overrides-for-test-hbase-site.xml", &c.

Reply:
IMHO, the tests as they were before you started were illegible and obtuse. I'm arguing that they should be easily understood by casual observation, not deep study. One way to achieve that is by well-considered, meaningful variable names and simple logic without lots of flow control.

Reply:
Agreed, was trying to avoid duplicate codes, let me update and upload a new patch.